_posts/2025-10-06-strimzi-metrics-reporter.md

```diff
@@ -0,0 +1,89 @@
+---
+layout: post
+title: "Using Strimzi Metrics Reporter to expose Kafka metrics in Prometheus format"
+date: 2025-10-06
+author: owen_corrigan
+---
+
+## Using Strimzi Metrics Reporter in your Kafka Cluster
+
+Apache Kafka generates and exposes metrics that reflect how it is operating.
+These metrics are useful for monitoring, troubleshooting, tuning, and capacity planning when it comes to running your Kafka cluster.
+In essence, monitoring is crucial to ensure the health and performance of your Kafka clusters.
+Prometheus is a metrics monitoring solution that is increasingly common due to its popularity in the cloud-native ecosystem, and it is this format that we use to expose metrics.
+Up to this point, we have recommended using the [JMX Exporter](https://github.com/prometheus/jmx_exporter) as a way to expose Kafka metrics, and we have provided some example YAML files in our [Strimzi Examples folder](https://github.com/strimzi/strimzi-kafka-operator/tree/main/packaging/examples/metrics).
+
+The [Strimzi Metrics Reporter](https://github.com/strimzi/metrics-reporter) is a new project in Strimzi, still in early access, and it is available since Strimzi 0.48.0 release.
+It implements the Kafka `MetricsReporter` interface to expose metrics via Prometheus.
+Strimzi Metrics Reporter is designed from scratch as a Kafka plugin and allows you to configure it directly from the Kafka configuration.
+That also means it is more efficient compared to other monitoring tools that run as sidecars or JVM agents.
+Strimzi Metrics Reporter has fixed names for all metrics giving users much less flexibility than other tools which provide complex mapping rules based on regular expressions.
+But it is easier to use and provides a metrics interface that is less fragile and fragmented, and more like an API.
+
+Here, we will discuss Strimzi Metrics Reporter configuration, and guide you on how to use it effectively to monitor your Kafka clusters.
+
+### Key Features
+* Native Prometheus support: The reporter exposes metrics in the Prometheus format through an HTTP endpoint, without relying on JMX.
+* Configurable Metrics Collection: Allows users to specify which metrics should be exposed using a flexible allowlist.
```

Comment:
A key feature is the performance and scalability. This is one of the main motivation from the [proposal](https://github.com/strimzi/proposals/blob/main/064-prometheus-metrics-reporter.md)

Reply:
Another one is better integration with Kafka. For example in the main branch of metrics-reporter we can update the server allowlist at runtime. That's still to be included in a release but it's a value-add over JMX Exporter.

Reply:
Actually, The JMX Exporter also updates the exposed metrics dynamically.

Reply:
Maybe:
* Scalability and Performance: Strimzi Metrics Reporter scales efficiently with increasing numbers of metrics, maintaining low response times even as the metric count grows, demonstrating strong performance characteristics and minimal overhead, making it well-suited for large-scale Kafka deployments.